src/cpp/desktop/DesktopWebPage.cpp

```diff
@@ -286,6 +286,11 @@ bool WebPage::acceptNavigationRequest(const QUrl &url,
    {
       return true;
    }
+   // allow iframes
+   else if (desktop::urlRequestInfoResourceType() == QWebEngineUrlRequestInfo::ResourceTypeSubFrame)
```

Comment:
If we're going to universally allow subframes to load, then we can dispense with almost every one of the above cases, because viewer URLs, Shiny apps, etc. are always loaded in subframes (even in dedicated windows they use an iframe wrapped in a GWT widget)

Reply:
Do you think this is reasonable now that we can sandbox our iframes?

Reply:
We can sandbox iframes, and we do sandbox the viewer pane when it's showing an external URL. 

https://github.com/rstudio/rstudio/blob/fc4cb83997bee8dda33cfaf4060dfaf0107f0ba7/src/gwt/src/org/rstudio/studio/client/workbench/views/viewer/ViewerPane.java#L263-L277

However this is largely irrelevant for this discussion as it's a fool's errand to try to guarantee that the `sandbox` attribute gets injected into every iframe RStudio ever loads. If we take this change we're basically putting our trust in QtWebEngine's cross domain security policy to keep us protected from whatever the wild internet throws at us.

Reply:
Got it (sorry for being a bit dense). I've updated this PR to ensure that we _don't_ load subframes (other than our own whitelisted set) and if we ever want to open the floodgates (so to speak) we can discuss that in a separate PR.

Reply:
Sounds good!